datastore/store.go

```diff
@@ -197,72 +197,26 @@ func (db *CachedDB) VRFNonce(id types.NodeID, epoch types.EpochID) (types.VRFPos
 	return nonce, nil
 }
 
-// GetAtxHeader returns the ATX header by the given ID. This function is thread safe and will return an error if the ID
+// GetAtx returns the ATX by the given ID. This function is thread safe and will return an error if the ID
 // is not found in the ATX DB.
-func (db *CachedDB) GetAtxHeader(id types.ATXID) (*types.ActivationTxHeader, error) {
+func (db *CachedDB) GetAtx(id types.ATXID) (*types.ActivationTx, error) {
 	if id == types.EmptyATXID {
 		return nil, errors.New("trying to fetch empty atx id")
 	}
 
-	if atxHeader, gotIt := db.atxHdrCache.Get(id); gotIt {
-		return atxHeader, nil
-	}
-
-	return db.getAndCacheHeader(id)
-}
-
-// GetFullAtx returns the full atx struct of the given atxId id, it returns an error if the full atx cannot be found
-// in all databases.
-func (db *CachedDB) GetFullAtx(id types.ATXID) (*types.ActivationTx, error) {
-	if id == types.EmptyATXID {
-		return nil, errors.New("trying to fetch empty atx id")
+	if atx, gotIt := db.atxCache.Get(id); gotIt {
+		return atx, nil
 	}
 
 	atx, err := atxs.Get(db, id)
 	if err != nil {
-		return nil, fmt.Errorf("get ATXs from DB: %w", err)
+		return nil, fmt.Errorf("get ATX from DB: %w", err)
 	}
 
-	db.atxHdrCache.Add(id, atx.ToHeader())
+	db.atxCache.Add(id, atx)
```

Comment:
This appears to be storing the ATX together with its blob in the cache. Wouldn't it substantially increase the memory use by the ATX cache?
(aside from the fact that `ActivationTx` by itself is quite a bit bigger than `ActivationTxHeader`)

Reply:
Ok, sounds good